### ISSUE TITLE
Several tests segfault when not using mpiexec

### ISSUE THREAD

**Issue by aancel** [feelpp/feelpp#563]
At least several tests seems to be ending up with a segfault when they are not launched with the `mpiexec` command.

Confirmed:
- testsuite/feelcore/feelpp_test_env
- testsuite/feelcore/feelpp_test_multithread

**Comment by aancel** [comment]
`gdb` doesn't seem to catch the error, but `valgrind` ends up with:    
```   
==65008== Process terminating with default action of signal 11 (SIGSEGV)
==65008==  Access not within mapped region at address 0xA98
==65008==    at 0x2DC6FE2F: ??? (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.55.0)
==65008==    by 0x314CA17F: ??? (in /lib/x86_64-linux-gnu/libc-2.19.so)
==65008==    by 0x3157C194: shutdown (syscall-template.S:81)
==65008==    by 0x47B3F2DD: mca_oob_tcp_fini (oob_tcp.c:1685)
==65008==    by 0x29C43D4B: mca_oob_base_close (oob_base_close.c:49)
==65008==    by 0x29C733BD: mca_base_components_close (mca_base_components_close.c:55)
==65008==    by 0x29C50E6E: orte_rml_base_close (rml_base_components.c:245)
==65008==    by 0x29C3422B: orte_ess_base_app_finalize (ess_base_std_app.c:244)
==65008==    by 0x4772E7DD: rte_finalize (ess_singleton_module.c:178)
==65008==    by 0x29C13A98: orte_finalize (orte_finalize.c:61)
==65008==    by 0x29BCF39A: ompi_mpi_finalize (ompi_mpi_finalize.c:430)
==65008==    by 0x2BD16751: boost::mpi::environment::~environment() (in /usr/lib/x86_64-linux-gnu/libboost_mpi.so.1.55.0)
==65008==  If you believe this happened as a result of a stack
==65008==  overflow in your program's main thread (unlikely but
==65008==  possible), you can try to increase the size of the
==65008==  main thread stack using the --main-stacksize= flag.
==65008==  The main thread stack size used in this run was 8388608.
```

**Comment by aancel** [comment]
The problem seems to be linked to GLOBAL_FIXTURE,
see http://boost.2283326.n4.nabble.com/Boost-test-and-openmpi-td4659414.html

**Comment by prudhomm** [comment]
Ok
does a global fixture fix this issue?

**Comment by aancel** [comment]
Apparently not, and the issue seems to be linked to this. The thread I linked does not provide a real solution to the problem, and the macros defined in Feelpp (i.e. FEELPP_ENVIRONMENT_WITH_OPTIONS, and FEELPP_ENVIRONMENT_NO_OPTIONS) use a global fixture that seems to cause the problem with mpi, at least openmpi. Users on the thread have reported it working with mpich.

I tried to modify the macros in `testsuite.hpp`: removing the deallocation of environment makes the problem disappear, but it is not a solution. So it is in the destructor that the problem appears and it seems to come from Boost mpi apparently. What puzzles me is that it works with mpirun but not without it.
